src/gui/spatial/index.js

```diff
@@ -35,10 +37,76 @@ realityEditor.gui.spatial.clearSpatialList = function (){
 };
 realityEditor.gui.spatial.lineAnimationList = {};
 
+realityEditor.gui.spatial.checkState = function() {
+    realityEditor.gui.spatial.historianOn = false;
+    realityEditor.gui.spatial.spatialOn = false;
+    
+    for (let ip in globalStates.spatial.whereIs) {
+        if (Object.keys(globalStates.spatial.whereIs[ip]).length > 0) {
+            realityEditor.gui.spatial.spatialOn = true;
+            break;
+        }
+    }
+
+    for (let ip in globalStates.spatial.howFarIs) {
```

Comment:
This is better, but if you want it to be really optimized this loop should be skipped if the previous loop set spatialOn to true, i.e. this:
```javascript
if (!realityEditor.gui.spatial.spatialOn) {
  for (let ip in globalStates.spatial.howFarIs) {
      if (Object.keys(globalStates.spatial.whereIs[ip]).length > 0) {
            realityEditor.gui.spatial.spatialOn = true;
            break;
      }
  }
}
```